### PR TITLE
Adds additional power actions

### PR DESCRIPTION
**Pull request by rodneyhbrown7**
Adds additional power actions for  restart now, power off now,  restart to F1 setup and restart the management controller.

Introduces a PowerManagement mixin to begin modularizing the client class. Moves power_ methods from client.rb to xclarity_power_management_mixin.rb.

### CODE REVIEW

File: docs/apib/node_bmc.apib [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FORMAT: 1A
+
+# XClarity Nodes
+
+# PUT /nodes/6B538D351B8211E193F65CF3FC6E4030/bmc
```
Comment (AndreyMenezes) [inline docs/apib/node_bmc.apib]: Put doesn't have any Json response?

Comment (rodneyhbrown7) [inline docs/apib/node_bmc.apib]: For the enabled test, I am only checking to see if the PUT request was actually made.  I can add a test that validates the response in some way. I will work on that.

File: lib/xclarity_client/node_management.rb [hunk 2]
```diff
@@ -33,5 +44,15 @@ def set_loc_led_state(uuid, state, name = 'Identify')
 
       do_put("#{Node::BASE_URI}/#{uuid}", request)
     end
+
+    private
+
+    def send_power_request(uri, requested_state = nil)
+      power_request = JSON.generate(powerState: requested_state)
+      response = do_put(uri, power_request)
+      msg = 'Power state action has been sent with request #{power_request}'
```
Comment (skovic) [inline lib/xclarity_client/node_management.rb]: I believe that string interpolation in Ruby only works with double quotes so please change this line to use double quotes.

Comment (AndreyMenezes) [inline lib/xclarity_client/node_management.rb]: I had not realized this. Double quotes is also the default in the project.

Comment (rodneyhbrown7) [inline lib/xclarity_client/node_management.rb]: Thanks @skovic that was a transcribe error.     I fixed that.   @AndreyMenezes , rubocop will complain if you use double quotes but don't have any interpolation between the double quotes.

File: docs/apib/node_bmc.apib (hunk 1 above)
Comment (rodneyhbrown7) [inline docs/apib/node_bmc.apib]: @AndreyMenezes I tested this again this morning.  There is no JSON response to the PUT request for these operations.

Comment (AndreyMenezes) [inline docs/apib/node_bmc.apib]: Ok

**Pull request merged**